<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RasterCalculatorDialogBase</class>
 <widget class="QDialog" name="RasterCalculatorDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Raster Calculator</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <!-- Expression Section -->
   <item>
    <widget class="QGroupBox" name="expressionGroup">
     <property name="title">
      <string></string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="expressionLayout">
      <!-- Top Section: Layers, Keypad, Functions -->
      <item>
       <layout class="QHBoxLayout" name="topLayout">
        <!-- Layers and Variables -->
        <item>
         <widget class="QGroupBox" name="layersGroup">
          <property name="title">
           <string>Layers and Variables</string>
          </property>
          <property name="minimumSize">
           <size>
            <width>160</width>
            <height>0</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="layersLayout">
           <item>
            <widget class="QListWidget" name="layersList">
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>180</height>
              </size>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <!-- Numeric Keypad and Operators -->
        <item>
         <widget class="QWidget" name="keypadWidget">
          <layout class="QGridLayout" name="keypadLayout">
           <property name="spacing">
            <number>3</number>
           </property>
           <!-- Row 0 -->
           <item row="0" column="0">
            <widget class="QPushButton" name="btn7">
             <property name="text"><string>7</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="btn8">
             <property name="text"><string>8</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="btn9">
             <property name="text"><string>9</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QPushButton" name="btnDiv">
             <property name="text"><string>/</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QPushButton" name="btnEq">
             <property name="text"><string>==</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="0" column="5">
            <widget class="QPushButton" name="btnNeq">
             <property name="text"><string>!=</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="0" column="6">
            <widget class="QPushButton" name="btnAnd">
             <property name="text"><string>&amp;</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <!-- Row 1 -->
           <item row="1" column="0">
            <widget class="QPushButton" name="btn4">
             <property name="text"><string>4</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="btn5">
             <property name="text"><string>5</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="btn6">
             <property name="text"><string>6</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QPushButton" name="btnMul">
             <property name="text"><string>*</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="1" column="4">
            <widget class="QPushButton" name="btnGt">
             <property name="text"><string>&gt;</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="1" column="5">
            <widget class="QPushButton" name="btnGte">
             <property name="text"><string>&gt;=</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="1" column="6">
            <widget class="QPushButton" name="btnOr">
             <property name="text"><string>|</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <!-- Row 2 -->
           <item row="2" column="0">
            <widget class="QPushButton" name="btn1">
             <property name="text"><string>1</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QPushButton" name="btn2">
             <property name="text"><string>2</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="btn3">
             <property name="text"><string>3</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QPushButton" name="btnSub">
             <property name="text"><string>-</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="2" column="4">
            <widget class="QPushButton" name="btnLt">
             <property name="text"><string>&lt;</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="2" column="5">
            <widget class="QPushButton" name="btnLte">
             <property name="text"><string>&lt;=</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="2" column="6">
            <widget class="QPushButton" name="btnPow">
             <property name="text"><string>^</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <!-- Row 3 -->
           <item row="3" column="0">
            <widget class="QPushButton" name="btn0">
             <property name="text"><string>0</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="btnDot">
             <property name="text"><string>.</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QPushButton" name="btnAdd">
             <property name="text"><string>+</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="3" column="3">
            <widget class="QPushButton" name="btnLParen">
             <property name="text"><string>(</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="3" column="4">
            <widget class="QPushButton" name="btnRParen">
             <property name="text"><string>)</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="3" column="5">
            <widget class="QPushButton" name="btnNot">
             <property name="text"><string>~</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
           <item row="3" column="6">
            <widget class="QPushButton" name="btnComma">
             <property name="text"><string>,</string></property>
             <property name="minimumSize"><size><width>32</width><height>26</height></size></property>
             <property name="maximumSize"><size><width>40</width><height>30</height></size></property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <!-- Functions List Panel -->
        <item>
         <widget class="QWidget" name="functionsWidget">
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>0</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="functionsOuterLayout">
           <property name="spacing">
            <number>2</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <!-- Function Category Combo -->
           <item>
            <widget class="QComboBox" name="functionCategoryCombo">
             <item>
              <property name="text"><string>Conditional</string></property>
             </item>
             <item>
              <property name="text"><string>Math</string></property>
             </item>
             <item>
              <property name="text"><string>Trigonometric</string></property>
             </item>
             <item>
              <property name="text"><string>Logical/Other</string></property>
             </item>
            </widget>
           </item>
           <!-- Functions List -->
           <item>
            <widget class="QListWidget" name="functionsList">
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <!-- Expression Text Area -->
      <item>
       <layout class="QHBoxLayout" name="expressionTextLayout">
        <item>
         <widget class="QPlainTextEdit" name="expressionEdit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>70</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>100</height>
           </size>
          </property>
          <property name="placeholderText">
           <string>Enter map algebra expression... (e.g., "dem@1" + 100)</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="expressionButtonsLayout">
          <item>
           <widget class="QPushButton" name="btnClear">
            <property name="text"><string>Clear</string></property>
            <property name="maximumSize"><size><width>60</width><height>16777215</height></size></property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnBackspace">
            <property name="text"><string>âŒ«</string></property>
            <property name="maximumSize"><size><width>60</width><height>16777215</height></size></property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <!-- Output Raster Section -->
   <item>
    <widget class="QGroupBox" name="outputGroup">
     <property name="title">
      <string>Output Raster</string>
     </property>
     <layout class="QHBoxLayout" name="outputLayout">
      <item>
       <widget class="QLineEdit" name="outputPathEdit">
        <property name="placeholderText">
         <string>Select output raster file path...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="browseBtn">
        <property name="text">
         <string>...</string>
        </property>
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="openOutputCheck">
        <property name="text">
         <string>Add to map</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <!-- Dialog Buttons -->
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="calculateBtn">
       <property name="text">
        <string>OK</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelBtn">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="helpBtn">
       <property name="text">
        <string>Help</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
